### DISCUSSION TITLE
Contributing

### DISCUSSION THREAD

**Discussion by bellrise** [opuntiaOS-Project/opuntiaOS#24] [General]
Hello,

I have a question regarding contributing to the project, specifically the custom `libc` implementation.
I saw the "development" status in the readme, so I wanted to ask if anything specifically needs some work
which I could write. Apart from `libc`, I'm also interested in writing code for other parts which run on C.

bellrise

**Comment by nimelehin** [comment]
Hey! Just opened this page for me :). If you still have interest you can work on `libc` or `libcxx`. There are many things to do there. On `libcxx` you can approach to implement `cin`. Feel free to open an issue to discuss a new feature you want to implement.

**Reply by nimelehin** [reply]
Oh, and now I am doing an experimental Objective-C runtime (which is almost on C), you can checkout branch `dev.objc` if you like

**Reply by bellrise** [reply]
Cool, I'll be sending a lot of PRs soon.

**Comment by bellrise** [comment]
Hey @nimelehin, I do have a question about forks. If I work on my own fork, and then I try to fetch & merge from this repository, it just makes a big mess in the commit history. Should I be making a new fork for each pull request, or is there a better way?

**Reply by nimelehin** [reply]
Hey! I recommend you to checkout a separate branch for each pr `git checkout -b` (google for more info). So you master will be a copy of upstream/master  (which is this project's master). Time to time you have to sync your master with upstream's master. Here is a [dock](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/syncing-a-fork) how to sync your master with upstream's

**Reply by bellrise** [reply]
Oh yeah that could work, so I keep master the same as the original repo and rebase when needed

**Reply by nimelehin** [reply]
Hey! I saw that you closed you first PR and made a new one. There is a better way to change your commits in PR: 
1) You can use `git rebase` to edit your commits.
2) After the edit do `force push` and both your branch on GitHub and PR will have edited commits.

**Reply by bellrise** [reply]
Yeah I figured it out, thanks

**Reply by nimelehin** [reply]
Could you join https://discord.gg/zxDcvkFAf5 please?
